db/repos/accounts.js

```diff
@@ -25,34 +30,311 @@ var columnSet;
  * @constructor
  * @return {AccountsRepo}
  */
-function AccountsRepo (db, pgp) {
-	this.db = db;
-	this.pgp = pgp;
-
-	this.dbTable = 'mem_accounts';
-
-	this.dbFields = [
-		'username',
-		'isDelegate',
-		'secondSignature',
-		'address',
-		'publicKey',
-		'secondPublicKey',
-		'balance',
-		'rate',
-		'rank'
-	];
-
-	if (!columnSet) {
-		columnSet = {};
-		var table = new pgp.helpers.TableName({table: this.dbTable, schema: 'public'});
-		columnSet.insert = new pgp.helpers.ColumnSet(this.dbFields, {table: table});
+class AccountsRepo {
+
+	constructor (db, pgp) {
+		this.db = db;
+		this.pgp = pgp;
+
+		this.dbTable = 'mem_accounts';
+
+		const ifNotExists = c => !c.exists;
+
+		// Used in SELECT, UPDATE, INSERT
+		const normalFields = [
+			{name: 'isDelegate', cast: 'int::boolean', skip: ifNotExists},
+			{name: 'u_isDelegate', cast: 'int::boolean', skip: ifNotExists},
+			{name: 'secondSignature', cast: 'int::boolean', skip: ifNotExists},
+			{name: 'u_secondSignature', cast: 'int::boolean', skip: ifNotExists},
+			{name: 'balance', cast: 'bigint', def: '0', skip: ifNotExists},
+			{name: 'u_balance', cast: 'bigint', def: '0', skip: ifNotExists},
+			{name: 'rate', cast: 'bigint', def: '0'},
+			{name: 'multimin', def: 0},
+			{name: 'u_multimin', def: 0},
+			{name: 'multilifetime', def: 0},
+			{name: 'u_multilifetime', def: 0},
+			{name: 'blockId', def: null},
+			{name: 'nameexist', def: 0},
+			{name: 'u_nameexist', def: 0},
+			{name: 'fees', cast: 'bigint', def: '0'},
+			{name: 'rewards', cast: 'bigint', def: '0'},
+			{name: 'vote', cast: 'bigint', def: '0'},
+			{name: 'producedblocks', cast: 'bigint', def: '0', prop: 'producedBlocks'},
+			{name: 'missedblocks', cast: 'bigint', def: '0', prop: 'missedBlocks'},
+			{name: 'username', def: null, skip: ifNotExists},
+			{name: 'u_username', def: null, skip: ifNotExists},
+			{name: 'publicKey', mode: ':raw', init: () => 'ENCODE("publicKey", \'hex\')', skip: ifNotExists},
```

Comment:
There is no property `mode` on the object, there is only `mod`, which is short for `modifier` ;)